Source/FieldSolver/SpectralSolver/SpectralAlgorithms/AvgGalileanAlgorithm.cpp

```diff
@@ -0,0 +1,371 @@
+        Array4<const Complex> Psi1_arr = Psi1_coef[mfi].array();
+        Array4<const Complex> Psi2_arr = Psi2_coef[mfi].array();
+        Array4<const Complex> Psi3_arr = Psi3_coef[mfi].array();
+        Array4<const Real> C1_arr = C1_coef[mfi].array();
+        Array4<const Real> S1_arr = S1_coef[mfi].array();
+        Array4<const Real> C3_arr = C3_coef[mfi].array();
+        Array4<const Real> S3_arr = S3_coef[mfi].array();
+
+
+        Array4<const Complex> A1_arr = A1_coef[mfi].array();
+        Array4<const Complex> A2_arr = A2_coef[mfi].array();
+        Array4<const Complex> Rhonew_arr = Rhonew_coef[mfi].array();
+        Array4<const Complex> Rhoold_arr = Rhoold_coef[mfi].array();
+        Array4<const Complex> Jcoef_arr =Jcoef_coef[mfi].array();
+        // Extract pointers for the k vectors
+        const Real* modified_kx_arr = modified_kx_vec[mfi].dataPtr();
+#if (AMREX_SPACEDIM==3)
+        const Real* modified_ky_arr = modified_ky_vec[mfi].dataPtr();
+#endif
+        const Real* modified_kz_arr = modified_kz_vec[mfi].dataPtr();
+
+        // Loop over indices within one box
+        ParallelFor(bx,
+        [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept
+        {
+            // Record old values of the fields to be updated
+            using Idx = SpectralAvgFieldIndex;
+
+            const Complex Ex_old = fields(i,j,k,Idx::Ex);
+            const Complex Ey_old = fields(i,j,k,Idx::Ey);
+            const Complex Ez_old = fields(i,j,k,Idx::Ez);
+            const Complex Bx_old = fields(i,j,k,Idx::Bx);
+            const Complex By_old = fields(i,j,k,Idx::By);
+            const Complex Bz_old = fields(i,j,k,Idx::Bz);
+
+            // Shortcut for the values of J and rho
+            const Complex Jx = fields(i,j,k,Idx::Jx);
+            const Complex Jy = fields(i,j,k,Idx::Jy);
+            const Complex Jz = fields(i,j,k,Idx::Jz);
+            const Complex rho_old = fields(i,j,k,Idx::rho_old);
+            const Complex rho_new = fields(i,j,k,Idx::rho_new);
+
+            const Complex Ex_avg = fields(i,j,k,Idx::Ex_avg);
+            const Complex Ey_avg= fields(i,j,k,Idx::Ey_avg);
+            const Complex Ez_avg = fields(i,j,k,Idx::Ez_avg);
+            const Complex Bx_avg = fields(i,j,k,Idx::Bx_avg);
+            const Complex By_avg = fields(i,j,k,Idx::By_avg);
+            const Complex Bz_avg = fields(i,j,k,Idx::Bz_avg);
+            // k vector values, and coefficients
+            const Real kx = modified_kx_arr[i];
+
+#if (AMREX_SPACEDIM==3)
+            const Real ky = modified_ky_arr[j];
+            const Real kz = modified_kz_arr[k];
+#else
+            constexpr Real ky = 0;
+            const Real kz = modified_kz_arr[j];
+#endif
+            constexpr Real c = PhysConst::c;
+            constexpr Real c2 = PhysConst::c*PhysConst::c;
+            constexpr Real inv_ep0 = 1._rt/PhysConst::ep0;
+            constexpr Complex I = Complex{0,1};
```

Comment:
Here I think it would be better to add the suffix _rt for single precision runs:
```suggestion
            constexpr Complex I = Complex{0._rt,1._rt};
```